macos/Sources/App/macOS/AppDelegate.swift

```diff
@@ -807,6 +807,12 @@ class AppDelegate: NSObject,
         setSecureInput(.toggle)
     }
 
+    @IBAction func toggleMaximize(_ sender: Any) {
```

Comment:
I think toggleMaximize should only affect terminal windows so putting this on `BaseTerminalController` and responding via first responder or via a notification seems like a better approach.

Today this doesn't matter much because all of our windows are terminal windows but that won't be true forever and if something like the command palette becomes a window one day (not tied to a specific terminal window) then it becomes a bigger issue and I'd rather not build debt for something simple here.